lib/chat_api_web/controllers/upload_controller.ex

```diff
@@ -0,0 +1,68 @@
+defmodule ChatApiWeb.UploadController do
+  use ChatApiWeb, :controller
+
+  alias ChatApi.Uploads
+  alias ChatApi.Uploads.Upload
+
+  action_fallback ChatApiWeb.FallbackController
+
+  def create(conn, %{
+        "upload" => upload_params,
+        "message_id" => message_id
```

Comment:
i'm a little confused about how this API should be used -- does a message need to be created before a file can be uploaded? (it seems like the "uploads" table requires a message_id?)

i wonder if a better model would be something like:
- a `files` table that basically just points at a resource in S3 and has some metadata
- an `attachments` table which handles mapping a `file` to a `message`

then, attaching a file to a message has two steps:
 1. upload a file to create a `file` record
 2. when a message is "sent", this causes the file to be "attached" to the message (in the form of creating an `attachment` record)

let me know if i'm overthinking it though

Reply:
I like the having the attachment being separate step since it is a clearer and it makes the API more explicit. But lets not use the word files since both Elixir and AWS has a concept of a file and it might be conflated. Happy to rename it to something else though.